### ISSUE TITLE
Wrong rest pagination calculation

### ISSUE THREAD

**Issue by mitalcoi** [Sylius/Sylius#8688]
Steps to reproduce:
1) Install fresh Sylius app (v1.0.0)
2) Send GET-query to fetch taxons with limit=2(!):
```sh
curl localhost:8000/api/v1/taxons/?limit=2 -H "Authorization: Bearer SampleToken" -H "Accept: application/json"
```
Pay attention on "_embedded:items" part of JSON, it contains only 1 element!
```
{
  "page": 1,
  "limit": 2,
  "pages": 4,
  "total": 7,
  "_links": {
    "self": {
      "href": "\/api\/v1\/taxons\/?page=1&limit=2"
    },
    "first": {
      "href": "\/api\/v1\/taxons\/?page=1&limit=2"
    },
    "last": {
      "href": "\/api\/v1\/taxons\/?page=4&limit=2"
    },
    "next": {
      "href": "\/api\/v1\/taxons\/?page=2&limit=2"
    }
  },
  "_embedded": {
    "items": [
      {
        "name": "Category",
        "id": 8,
        "code": "category",
        "position": 0,
        "translations": {
          "en_GB": {
            "locale": "en_GB",
            "id": 15,
            "name": "Category",
            "slug": "category",
            "description": "Et illo nobis pariatur itaque. Accusantium voluptatem autem voluptates quas. Ut suscipit quia rerum et."
          },
          "en_US": {
            "locale": "en_US",
            "id": 16,
            "name": "Category",
            "slug": "category",
            "description": "Et illo nobis pariatur itaque. Accusantium voluptatem autem voluptates quas. Ut suscipit quia rerum et."
          }
        },
        "images": [
          
        ],
        "_links": {
          "self": {
            "href": "\/api\/v1\/taxons\/category"
          }
        }
      }
    ]
  }
}
```

**Comment by lchrusciel** [comment]
I'm afraid it can be a problem with hydration rather than with pagination. The paginator part of request seems to be calculated properly
```json
{
  "page": 1,
  "limit": 2,
  "pages": 4,
  "total": 7,
  "_links": {
    "self": {
      "href": "\/api\/v1\/taxons\/?page=1&limit=2"
    },
    "first": {
      "href": "\/api\/v1\/taxons\/?page=1&limit=2"
    },
    "last": {
      "href": "\/api\/v1\/taxons\/?page=4&limit=2"
    },
    "next": {
      "href": "\/api\/v1\/taxons\/?page=2&limit=2"
    }
  }
}
```

**Comment by samvdb** [comment]
I just bumped into the same problem. Setting fetchJoinCollection to `true` fixed the issue for me, however there was a reason why this was set to false...
I forked grid bundle and made the fix there.
https://github.com/samvdb/SyliusGridBundle/commit/cdcea43c67a1aa26991ac3c1dc486727b015e220

Maybe @lchrusciel has an idea?

Relevant commit was:
https://github.com/Sylius/SyliusGridBundle/commit/f8beacde66b921b6f0c94d7ea4a8928413904a8b
